### PR TITLE
Contributing and tests starter

### PR DESCRIPTION
**Pull request by aug2uag**
making a pull request without running clippy and tests can lead to fail

this PR adds a contributing document with a point of entry and structure for scripts management of testing

### CODE REVIEW

**Comment by tarcieri** [comment]
I don't think it makes sense to add Python scripts to this project.

A Makefile would be ok.

**Comment by aug2uag** [comment]
I think a Makefile is overkill for this no? Perhaps a bash script or external support package "cargo-audit-tests" for the purpose of ensuring contributors run the tests, and to facilitate installing clippy, format, and any others that come up in the future. Please let me know what you think of this.

**Comment by tarcieri** [comment]
Personally I think anything more than CONTRIBUTING.md is overkill for this: running tests, rustfmt, and clippy is pretty standard for a Rust project, and in the event you don't run one of them locally, CI will catch it.

**Comment by aug2uag** [comment]
Thank you Tony! What do you feel about adding them to tests/config.rs or are there any necessary tests missing there?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+2. make changes
+
+3. run tests
+- a) `cargo test`
+- b) `cargo clippy`
+- c) `cargo rustfmt`
+
+4. commit format
+- TODO: commit message formatting needed
```
Comment (tarcieri) [inline CONTRIBUTING.md]: This repo hasn't had an official commit message policy. Perhaps we should adopt one.

**Comment by tarcieri** [comment]
I will try to write some contributing guidelines in the future

**Pull request closed without merging**